reports/tests/test_msisdn_message_report.py

```diff
@@ -1,60 +1,679 @@
+                "start_date": '2017-01-01',
+                "end_date": '2018-01-01',
+                "email_subject": 'The Email Subject',
+                "status": ReportTaskStatus.PENDING
+            })
+
+            generate_msisdn_message_report.apply_async(kwargs={
+                'start_date': '2017-01-01',
+                'end_date': '2018-01-01',
+                'email_recipients': ['foo@example.com'],
+                'email_subject': 'The Email Subject',
+                'task_status_id': task_status.id,
+                'msisdns': msisdns
+            })
+
+            return filename
+
+    @responses.activate
+    @mock.patch("os.remove")
+    def test_generate_report_email(self, mock_remove):
+        """
+        Generating a report should create an email with the correct address,
+        subject, and attachment.
+        """
+
+        self.add_response_identity_store_search('%2B2340000000', [])
+        self.trigger_report_generation(['+2340000000'])
+        [report_email] = mail.outbox
+        self.assertEqual(report_email.subject, 'The Email Subject')
+        (file_name, data, mimetype) = report_email.attachments[0]
+        self.assertEqual('msisdn-report-2017-01-01-to-2018-01-01.xlsx',
+                         file_name)
+
+    @responses.activate
+    def test_generate_report_status_done(self):
+        """
+        Generating a report should mark the ReportTaskStatus objects as Done if
+        it is successful.
+        """
+
+        self.add_response_identity_store_search('%2B2340000000', [])
+        self.trigger_report_generation(['+2340000000'])
+
+        task_status = ReportTaskStatus.objects.last()
+        self.assertEqual(task_status.status, ReportTaskStatus.DONE)
+        self.assertEqual(task_status.file_size > 5000, True)
+
+    @mock.patch("reports.tasks.send_email.SendEmail.apply_async")
+    @responses.activate
+    def test_generate_report_status_email(self, mock_send):
+        """
+        Generating a report should mark the ReportTaskStatus objects as Sending
+        before it sends the email.
+        """
+
+        self.add_response_identity_store_search('%2B2340000000', [])
+        self.trigger_report_generation(['+2340000000'])
+
+        task_status = ReportTaskStatus.objects.last()
+        self.assertEqual(task_status.status, ReportTaskStatus.SENDING)
+        self.assertEqual(task_status.file_size > 5000, True)
+
+    @responses.activate
+    @mock.patch("reports.tasks.send_email.SendEmail.apply_async")
+    def test_generate_report_status_running(self, mock_send):
+        """
+        Generating a report should mark the ReportTaskStatus objects as Running
+        while the generation is in progress.
+        """
+
+        orig = openpyxl.Workbook.save
+
+        def new_save(wb, file):
+            task_status = ReportTaskStatus.objects.last()
+            self.assertEqual(task_status.status, ReportTaskStatus.RUNNING)
```

Comment:
Since we're doing the assert in here, if `openpyxl.Workbook.save` is never called, this test will pass successfully. I think we either need to assert that the mock was run, or have a variable that changes inside this function so that we can assert that this has been run

Reply:
:face_with_head_bandage: :+1: